File: Core/Entities/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities
{
    public class Appointment : BaseEntity
    {
        public int? PatientId { get; set; }     
        [ForeignKey("PatientId")]
        public Patient Patient { get; set; } 

        public int Office1d { get; set; }     
        [ForeignKey("OfficeId")]
        public Office Office { get; set; }     


        [DataType(DataType.Date)]
        public DateTime StartDateAndTimeOfAppointment { get; set; }

        [DataType(DataType.Date)]
        public DateTime EndDateAndTimeOfAppointment { get; set; }

        public bool? Status { get; set; }
        public string Remarks { get; set; }



    }
}

File: angular-edoctor/src/app/shared/models/appointment.ts
export interface Appointment {
    id: number;
    doctor: string;
    patientId: number;
    doctorId: number;
    officeId: number;
    appointmentId: number;
    patient: string;
    country: string;
    officeAddress: string;
    city: string;
    hospital: string;
    startDateAndTimeOfAppointment: Date;
    endDateAndTimeOfAppointment: Date;
    status: boolean;
    remarks: string;
}

export class AppointmentCreateEdit {
    id: number;
    officeId: number;
    patientId: number;
    startDateAndTimeOfAppointment: Date;
    endDateAndTimeOfAppointment: Date;
    remarks: string;
    status: string;
}
